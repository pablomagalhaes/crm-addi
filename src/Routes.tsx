import React, { FC } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import { toLoadable } from '@/utils';
import Layout from './components/layout/Layout.component';

// Lazy Loading Views
const IndexView = toLoadable(() => import('@/views/index/Index.view'));

const Redirect404View = toLoadable(
  () => import('@/views/404/Redirect404.view')
);

const Routes: FC = () => (
  <Router>
    <Layout>
      <Switch>
        <Route exact path="/" component={IndexView} />

        <Route component={Redirect404View} />
      </Switch>
    </Layout>
  </Router>
);

export default Routes;
