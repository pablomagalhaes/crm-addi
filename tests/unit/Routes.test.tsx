import React from 'react';
import { shallow } from 'enzyme';

import Routes from '@/Routes';

describe('<Routes />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
});
