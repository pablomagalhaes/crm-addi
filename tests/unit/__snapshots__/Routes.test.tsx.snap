// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should match snapchat 1`] = `
<BrowserRouter>
  <Layout>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </Layout>
</BrowserRouter>
`;
